src/urllib3/util/ssl_.py

```diff
@@ -121,8 +121,9 @@ def assert_fingerprint(cert, fingerprint):
     cert_digest = hashfunc(cert).digest()
 
     if not hmac.compare_digest(cert_digest, fingerprint_bytes):
+        got_fingerprint = hexlify(cert_digest).decode()
         raise SSLError(
-            f'Fingerprints did not match. Expected "{fingerprint}", got "{hexlify(cert_digest)}".'
```

Comment:
We already have `test/with_dummyserver/test_https.py::TestHTTPS::test_assert_invalid_fingerprint`

Reply:
Oh, but didn't behavior change here from `hexlify(...)` to `hexlify(...).decode()`? If behavior changed I'm curious why the test status didn't change.

It looks like the tests simply check for "Fingerprints did not match" but don't test the appearance of the rest of the message. Could we update them to add the rest of the message as well?

Reply:
Huh, I thought the only change was about the warning, but inded, behavior changed. Before that fix we had a b'' marker that we never noticed because we didn't look at the full string in the test. I now fixed that, thanks!